src/clientfactory.ts

```diff
@@ -42,11 +42,18 @@ export class DiscordClientFactory {
         this.botClient = new DiscordClient({
             fetchAllMembers: true,
             messageCacheLifetime: 5,
-            sync: true,
+        });
+
+        const waitPromise = new Promise((resolve, reject) => {
+            this.botClient.once("shardReady", resolve);
+            this.botClient.once("shardError", reject);
         });
 
         try {
-            await this.botClient.login(this.config.botToken);
+            await this.botClient.login(this.config.botToken, true);
```

Comment:
You might want to store if tokens are user or bot tokens in the db. After a ready you can fetch your own status and store that in the db for existing ones set without explicitly specifying if it is bot or user

Reply:
We only ever store user tokens in the DB. The only bot token is in the config. If we ever changed this we would add appropriate DB stuffs.

Reply:
Actually, a user might want to puppet a bot token.

Reply:
Yes, I think that falls under "If we ever changed this we would add appropriate DB stuffs."